.github/workflows/continuous-integration.yml

```diff
@@ -25,9 +25,27 @@ on: ["push", "pull_request"]
 jobs:
   # Thanks to github user @martyrs
   # https://github.community/t/how-to-properly-clean-up-self-hosted-runners/128909/3
-  cleaner:
-    name: Delete Self-Hosted Runner Workspaces
-    runs-on: self-hosted
```

Comment:
Stupid question.  If we leave this as `self-hosted` won't it run on all machines that are tagged as self-hosted?  Then we don't need to duplicate this code three times

Reply:
Oh, I see... what you did forces it to run on all three machines.

I think an easier way may be to convert this to an action and then invoke it as follows from `functionaltestsv2_cgroup_legacy`, `functionaltestsv2_cgroup_hybrid`, etc.:

```
- name: Clean the workspace
  uses: ./.github/actions/clean-workspace
```

Reply:
One more random thought.  It looks like Github added the ability to run a script before or after a job.  They specifically cite a cleanup example
https://docs.github.com/en/actions/hosting-your-own-runners/running-scripts-before-or-after-a-job

Reply:
@drakenclimber Thanks for the great suggestions. I tried both of your suggestions but was not successful with either of them :(.  The `run a script` approach, is not documented well enough for me :( and I tried different combinations and ideas to export/trigger `ACTIONS_RUNNER_HOOK_JOB_STARTED` but was not able to get it working. 
The other approach of using `./.github/actions/clean-workspace` has one issue, the git is cloned once and every action is checked out from the cloned tree, so running the action to delete the workspace will delete the tree, and hence the tree is not available for other steps, ending up with the failed job. I also tried using the cleanup action as the last step, thinking it will safe to do so, as we do not need the tree any more and it too failed.
I have another idea, of integrating the `cleaner` step, as the first one to run, something like below:
```

Reply:
I got it working on our runners.  I couldn't get the `.env` file method to work, but it did work by setting the environment variable `$ACTIONS_RUNNER_HOOK_JOB_STARTED` to point to a script to clean the workspace.

Here's the script I'm using
```
#!/bin/bash

echo "Deleting $GITHUB_WORKSPACE" | sudo tee /dev/kmsg
rm -rf $GITHUB_WORKSPACE
mkdir -p $GITHUB_WORKSPACE
```